tools/performance/parse_perf_test.py

```diff
@@ -45,10 +45,23 @@ def run(i: int):
     log.info('Complete in %0.2fs, using %d KB', duration, mem)
     return duration, mem
 
+
 def parse_time_output(output):
     parts = output.split(" ")
     return float(parts[0].strip()), int(parts[1].strip())
 
+
+def read_cpu_info():
+    """Return the CPU model number & number of CPUs."""
+    try:
+        with open('/proc/cpuinfo') as f:
```

Comment:
Do we get virtualised machines on circle ci? I was trying to see if we could get more fine grained information from it but could only find a reference in https://circleci.com/docs/2.0/configuration-reference/, `Note: If you want to confirm how much memory you have been allocated, you can check the cgroup memory hierarchy limit with grep hierarchical_memory_limit /sys/fs/cgroup/memory/memory.stat.` I wonder if the same could be retrieved for the allocated cpu.

Reply:
Yeah they are virtualised within Docker (you can also have VM runners but we don't).

It's possible the cgroup setup could be relevant too but that should just match the machine type we've requested; the info here should still be useful (e.g. Go will determine GOMAXPROCS this way, I don't think it knows about cgroups).